### ISSUE TITLE
demo-app crashes when minimize window on Windows 10

### ISSUE THREAD

**Issue by maidamai0** [nmwsharp/polyscope#84]
demo app crashes when minimized on Windows 10.
the crash is caused by an exception on: [src/render/engine.cpp:270](https://github.com/nmwsharp/polyscope/blob/8e464a790e8604001d57ee13368e8c889145b299/src/render/engine.cpp#L262)
It can be avoided by a small check but I think there may be a better way to handle this:
```cpp
void Engine::applyLightingTransform(std::shared_ptr<TextureBuffer>& texture) {
  // compute downsampling rate
  glm::vec4 currV = getCurrentViewport();



  // avoid devide by zero
  if(currV[2] <= std::numeric_limits<float>::epsilon() || currV[3] <= std::numeric_limits<float>::epsilon()) {
    std::cout << currV << std::endl;  // outputs <0,0,0>
    return;
  }



  float sampleX = texture->getSizeX() / currV[2];
  float sampleY = texture->getSizeY() / currV[3];
  if (sampleX != sampleY) throw std::runtime_error("lighting downsampling should have same aspect");
...
return;
```

**Comment by evouga** [comment]
I am encountering the same crash.

If it helps I am using MSVC 2019 with Imgui compiled in GLFW mode.

**Comment by maidamai0** [comment]
I use *MSVC 2019 with Imgui compiled in GLFW mode* too.

**Comment by nmwsharp** [comment]
Uh oh, that sounds bad. Thank you for raising this @maidamai0 and @evouga for confirming. I'll look in to it ASAP.

**Comment by nmwsharp** [comment]
OK, I reproduced this on my end as well.

I poked around the GLFW windowing code and don't see any obvious errors, and I'm not sure why this only shows up on Windows.... We could probably handle this explicitly by explicitly checking minimize state, but that's a potentially deeper change since we still want to allow offscreen rendering where possible (e.g. for making movies, though I don't think offscreen rendering works currently).

In lieu of a "real" solution, I pushed a simple fix to address the symptom in 5138ee846c435b051e4dbabd6c7d7f56bb7c0452. This resolves the problem on my machine. 

Please feel free to reopen the issue if you are still seeing problems.
